fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/GroupByCountDistinctRewriteRule.java

```diff
@@ -198,19 +189,41 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
                 OptExpression.create(second.build(), OptExpression.create(first, input.getInputs())));
     }
 
-    @NotNull
-    private CallOperator genAggregation(String name, Type[] argTypes, List<ScalarOperator> args, CallOperator origin) {
-        if (FunctionSet.SUM.equals(name) || FunctionSet.MAX.equals(name) || FunctionSet.MIN.equals(name)) {
-            // for decimal
-            return new CallOperator(name, origin.getType(), args, origin.getFunction());
+    private CallOperator transformOtherAgg(CallOperator origin, List<ScalarOperator> args, boolean isFirst) {
+        String originFuncName = origin.getFunction().functionName();
+        if (isFirst) {
+            String firstFuncName = OTHER_FUNCTION_TRANS.get(originFuncName).first;
+            if (StringUtils.equals(originFuncName, firstFuncName)) {
+                return origin;
+            } else {
+                Function newFunc = Expr.getBuiltinFunction(firstFuncName, origin.getFunction().getArgs(),
+                        Function.CompareMode.IS_NONSTRICT_SUPERTYPE_OF);
+                return new CallOperator(firstFuncName, newFunc.getReturnType(), args, newFunc);
+            }
+        } else {
+            String secondFuncName = OTHER_FUNCTION_TRANS.get(originFuncName).second;
+            Type[] argTypes = args.stream().map(ScalarOperator::getType).toArray(Type[]::new);
+            Function newFunc = origin.getFunction().updateArgType(argTypes);
```

Comment:
update args type may lead to using the wrong method, it is better to use `Expr.getBuiltinFunction`?